i18n-helpers/src/bin/mdbook-gettext.rs

```diff
@@ -30,17 +30,64 @@ use mdbook::preprocess::{CmdPreprocessor, PreprocessorContext};
 use mdbook::BookItem;
 use mdbook_i18n_helpers::{extract_events, reconstruct_markdown, translate_events};
 use polib::catalog::Catalog;
+use polib::message::Message;
 use polib::po_file;
+use pulldown_cmark::Event;
 use semver::{Version, VersionReq};
 use std::{io, process};
 
+/// Strip formatting from a Markdown string.
+///
+/// The string can only contain inline text. Formatting such as
+/// emphasis and strong emphasis is removed.
+///
+/// Modelled after `mdbook::summary::stringify_events`.
+fn strip_formatting(text: &str) -> String {
+    extract_events(text, None)
+        .iter()
+        .filter_map(|(_, event)| match event {
+            Event::Text(text) | Event::Code(text) => Some(text.as_ref()),
+            Event::SoftBreak => Some(" "),
+            _ => None,
+        })
+        .collect()
+}
+
 fn translate(text: &str, catalog: &Catalog) -> String {
     let events = extract_events(text, None);
     let translated_events = translate_events(&events, catalog);
     let (translated, _) = reconstruct_markdown(&translated_events, None);
     translated
 }
 
+/// Update `catalog` with stripped messages from `SUMMARY.md`.
+///
+/// While it is permissible to include formatting in the `SUMMARY.md`
+/// file, `mdbook` will strip it out when rendering the book. It will
+/// also strip formatting when sending the book to preprocessors.
+///
+/// To be able to find the translations for the `SUMMARY.md` file, we
+/// append versions of these messages stripped of formatting.
+fn add_stripped_summary_translations(catalog: &mut Catalog) {
+    let mut stripped_messages = Vec::new();
+    for msg in catalog.messages() {
+        if !msg.source().contains("SUMMARY.md") {
```

Comment:
Perhaps `ends_width` instead of `contains`?

Reply:
Ah, that would be nice, but we're actually matching against the full list of filenames here: something like `src/foo/bar.md:123 src/SUMMARY.md:234`.

I could parse this and it would look something like this:

```rust
        if !msg
            .source()
            .split_whitespace()
            .filter_map(|source| source.split_once(':'))
            .map(|(path, _)| Path::new(path))
            .any(|path| path.file_name() == Some(OsStr::new("SUMMARY.md")))
        {
            continue;
        }
```

However, now that I just tried this out, I realize that we cannot make any strong assumptions on where the `SUMMARY.md` file is: `mdbook` allows you to configure the [source directory](https://rust-lang.github.io/mdBook/format/configuration/general.html#general-metadata).

So the code above is essentially the same as just checking if `SUMMARY.md` is found anywhere in the string.